### ISSUE TITLE
`scale` in `cgrad` doesn't work with CairoMakie heatmap

### ISSUE THREAD

**Issue by Moelf** [JuliaPlots/Makie.jl#2040]
```julia
x = repeat(1:10)
y = 1:10
z = (2) .^ (1:10)
f,a,p = heatmap(x,y,z; colormap=cgrad(:default, scale=:exp))
Colorbar(f[:, 2], p)
f
```
![image](https://user-images.githubusercontent.com/5306213/173120562-99867df6-e662-4e18-9060-e587c4a33a62.png)

```julia
f,a,p = heatmap(x,y,z; colormap=cgrad(:default, scale=:log10))
Colorbar(f[:, 2], p)
f
```
![image](https://user-images.githubusercontent.com/5306213/173120662-b6cc1c76-452c-47f1-ae67-da5a21d326c0.png)

as you can see the colorbar changed but the rendered heatmap doesn't have visual different?

cc. @mivanovska007

**Comment by asinghvi17** [comment]
Which version of Makie are you using?  IIRC this was fixed recently.

**Comment by Moelf** [comment]
indeed it's fixed in 0.17.4
